### PR TITLE
Feature 201 timeline editor screen and overall content areas

### LINKED ISSUE #201: Timeline editor screen and overall content areas

### PR DESCRIPTION
**Pull request by terolamminpaa**
resolves #201

### CODE REVIEW

File: src/components/app.tsx [hunk 1]
```diff
@@ -165,6 +166,14 @@ class App extends React.Component<Props, State> {
...
+                        //   history={ history }
+                        //   exhibitionId={ match.params.exhibitionId }
+                        //   floorId={ match.params.floorId }
+                        //   roomId={ match.params.roomId }
+                        //   contentVersionId={ match.params.contentVersionId }
+                        //   groupContentVersionId={ match.params.groupContentVersionId }
+                        // />
                         <TimelineScreen
                           history={ history }
                           exhibitionId={ match.params.exhibitionId }
```
Comment (jarinykanen) [inline src/components/app.tsx:179]: Revert

File: src/components/generic/content-utility-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  noTabs?: boolean;
+  history: History;
+  actionBarButtons?: ActionButton[];
+}
+
+/**
+ * Interface representing component state
+ */
+interface State {
+
```
Comment (jarinykanen) [inline src/components/generic/content-utility-bar.tsx:23]:
```suggestion
```

File: src/components/generic/device-list.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * Interface representing component state
+ */
+interface State {
+  selectedDevice?: ExhibitionDevice;
+}
+
+/**
+ * Component for content utility bar
+ */
+class TabList extends React.Component<Props, State> {
```
Comment (jarinykanen) [inline src/components/generic/device-list.tsx:26]: Fix docs

File: src/components/generic/device-list.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  /**
+   * Constructor
+   *
+   * @param props component properties
+   */
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+
+    };
```
Comment (jarinykanen) [inline src/components/generic/device-list.tsx:37]:
```suggestion
    this.state = {
    };
```

File: src/components/generic/device-list.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        dense
+        className={ classes.tabs }
+      >
+        { this.renderTabButtons() }
+      </List>
+    );
+  }
+
+  /**
+   * Renders tab button
```
Comment (jarinykanen) [inline src/components/generic/device-list.tsx:58]:
```suggestion
   * Renders tab buttons
```

File: src/components/generic/editor-utility-bar.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  actionBarButtons?: ActionButton[];
+  setSelectedDevice?: (deviceId: string) => ExhibitionDevice | undefined;
+}
+
+/**
+ * Interface representing component state
+ */
+interface State {
+
+}
```
Comment (jarinykanen) [inline src/components/generic/editor-utility-bar.tsx:25]:
```suggestion
interface State {
}
```

File: src/components/generic/editor-utility-bar.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+   * Constructor
+   *
+   * @param props component properties
+   */
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+
+    };
+  }
```
Comment (jarinykanen) [inline src/components/generic/editor-utility-bar.tsx:42]:
```suggestion
    this.state = {
    };
  }
```

File: src/components/generic/tab-list.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  history: History;
+  tabButtons: NavigationButton[];
+}
+
+/**
+ * Interface representing component state
+ */
+interface State {
+
+}
```
Comment (jarinykanen) [inline src/components/generic/tab-list.tsx:21]:
```suggestion
interface State {
}
```

File: src/components/generic/tab-list.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+interface NavigationButton {
```
Comment (jarinykanen) [inline src/components/generic/tab-list.tsx:23]: Add docs

File: src/components/generic/tab-list.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  postfix: string;
+  text: string;
+}
+
+/**
+ * Component for content utility bar
+ */
+class TabList extends React.Component<Props, State> {
```
Comment (jarinykanen) [inline src/components/generic/tab-list.tsx:31]: Fix docs

File: src/components/generic/tab-list.tsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+    );
+  }
+
+  /**
+   * Renders tabs
+   */
+  private renderTabs = () => {
+    const { classes } = this.props;
+    
+
```
Comment (jarinykanen) [inline src/components/generic/tab-list.tsx:61]:
```suggestion

```

File: src/components/generic/tab-list.tsx [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+      );
+    });
+  }
+
+  /**
+   * Get new path.
+   *
+   * @param tabButton navigation button
+   */
+  private getTabButtonPath = (tabButton: NavigationButton): string => {
```
Comment (jarinykanen) [inline src/components/generic/tab-list.tsx:104]: Add return docs

File: src/components/generic/top-bar.tsx [hunk 13]
```diff
@@ -19,19 +21,22 @@ interface Props extends WithStyles<typeof styles> {
...
   onDashboardButtonClick?: () => void;
+  setSelectedDevice?: (deviceId: string) => ExhibitionDevice | undefined;
 }
 
 /**
  * Interface representing component state
  */
 interface State {
-  navigationButtons: NavigationButton[];
+
```
Comment (jarinykanen) [inline src/components/generic/top-bar.tsx:39]:
```suggestion
```

File: src/components/layouts/element-timeline-pane.tsx [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+import { WithStyles, withStyles } from "@material-ui/core";
+import styles from "../../styles/element-timeline-pane";
+
+
+/**
+ * Interface representing component properties
+ */
+interface Props extends WithStyles<typeof styles> {
+
```
Comment (jarinykanen) [inline src/components/layouts/element-timeline-pane.tsx:11]:
```suggestion
```

File: src/components/layouts/element-timeline-pane.tsx [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+/**
+ * Interface representing component state
+ */
+interface State {
+
```
Comment (jarinykanen) [inline src/components/layouts/element-timeline-pane.tsx:18]:
```suggestion
```

File: src/components/layouts/element-timeline-pane.tsx [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+/**
+ * Component for element contents pane
+ */
+class ElementTimelinePane extends React.Component<Props, State> {
```
Comment (jarinykanen) [inline src/components/layouts/element-timeline-pane.tsx:24]: Fix docs

File: src/components/screens/content-editor-screen.tsx [hunk 17]
```diff
@@ -0,0 +1,392 @@
...
+import ElementSettingsPane from "../layouts/element-settings-pane";
+import ElementContentsPane from "../layouts/element-contents-pane";
+import ElementNavigationPane from "../layouts/element-navigation-pane";
+import PagePreview from "../preview/page-preview";
+import produce from "immer";
+import CodeEditor from "../editor/code-editor";
+import AndroidUtils from "../../utils/android-utils";
+import PanZoom from "../generic/pan-zoom";
+type View = "CODE" | "VISUAL";
+
```
Comment (jarinykanen) [inline src/components/screens/content-editor-screen.tsx:27]:
```suggestion
```

File: src/components/screens/content-editor-screen.tsx [hunk 18]
```diff
@@ -0,0 +1,392 @@
...
+  /**
+   * Component did mount life cycle handler
+   */
+  public componentDidMount = async () => {
+    const { exhibition, exhibitionId, accessToken } = this.props;
+
+    if (!exhibition || exhibitionId !== exhibition.id) {
+      const exhibitionsApi = Api.getExhibitionsApi(accessToken);
+      this.props.setSelectedExhibition(await exhibitionsApi.findExhibition({ exhibitionId }));
+      return;
```
Comment (jarinykanen) [inline src/components/screens/content-editor-screen.tsx:92]:
```suggestion
    const { exhibition, exhibitionId, accessToken, setSelectedExhibition } = this.props;

    if (!exhibition || exhibitionId !== exhibition.id) {
      const exhibitionsApi = Api.getExhibitionsApi(accessToken);
      setSelectedExhibition(await exhibitionsApi.findExhibition({ exhibitionId }));
      return;
```

File: src/components/app.tsx (hunk 1 above)
Comment (terolamminpaa) [reply src/components/app.tsx]: Reverted

**Pull request merged**